src/core/components/collapsible/collapsible.tsx

```diff
@@ -30,9 +24,26 @@ export const Collapsible: React.FC<ICollapsibleProps> = (props) => {
     const [maxHeight, setMaxHeight] = useState(0);
 
     const isOpen = isOpenProp ?? isOpenState;
-
     const contentRef = useRef<HTMLDivElement>(null);
-    const maxCollapsedHeight = customCollapsedHeight ?? sizedCollapsedHeights[collapsedSize];
+
+    const calculateCollapsedHeight = useCallback(() => {
+        if (collapsedPixels != null) {
+            return collapsedPixels;
+        }
+
+        const defaultLineHeight = 24;
+        if (collapsedLines && contentRef.current) {
+            const lineHeight = parseFloat(window.getComputedStyle(contentRef.current).lineHeight);
+
+            if (!Number.isNaN(lineHeight)) {
+                return lineHeight * collapsedLines;
+            }
+        }
+
+        return collapsedLines * defaultLineHeight;
+    }, [collapsedPixels, collapsedLines]);
+
+    const [collapsedHeight, setCollapsedHeight] = useState(calculateCollapsedHeight());
```

Comment:
I'm not sure we need state here, result of `calculateCollapsedHeight` does not change when ResizeObserver triggers.

I think we could just calculate value straight away, like this:

```suggestion
    const collapsedHeight = calculateCollapsedHeight();
```